frontend/compiled/term.go

```diff
@@ -175,3 +171,26 @@ func (t Term) string(sbb *strings.Builder, coeffs []big.Int) {
 	}
 	sbb.WriteString(strconv.Itoa(t.WireID()))
 }
+
+func (t Term[E, ptE]) HashCode() uint64 {
+	return t.Var
+}
+
+func (t Term[E, ptE]) IsZero() bool {
+	return ptE(&t.Coeff).IsZero()
+}
+
+func (t Term[E, ptE]) IsOne() bool {
+	return ptE(&t.Coeff).IsOne()
+}
+
+func (t Term[E, ptE]) IsNegOne() bool {
```

Comment:
this probably does 1 heap alloc per call right?

Reply:
Yup, does allocate. Due to type parameter cannot init once easily. In many cases we use this method in the constraint system, I guess could store -1 there?